src/cdk/tree/control/nested-tree-control.ts

```diff
@@ -26,15 +26,17 @@ export class NestedTreeControl<T> extends BaseTreeControl<T> {
   expandAll(): void {
     this.expansionModel.clear();
     let toBeExpanded = <any>[];
-    this.dataNodes.forEach(dataNode => toBeExpanded.push(...this.getDescendants(dataNode)));
+    this.dataNodes.forEach(dataNode =>
+        toBeExpanded.push(...this.getDescendants(dataNode), dataNode));
     this.expansionModel.select(...toBeExpanded);
   }
 
   /** Gets a list of descendant dataNodes of a subtree rooted at given data node recursively. */
   getDescendants(dataNode: T): T[] {
     const descendants = [];
     this._getDescendants(descendants, dataNode);
-    return descendants;
+    // Remove the node itself
```

Comment:
This only returns a slice with the second item in it, but it doesn't remove it. Is it intended?